### PR TITLE
Add prerequisites and provisioning

### LINKED ISSUE #10355: Document new way of installation
**Description**

Kyma cli will offer new way of installation `kyma deploy` which needs to be documented alongside the "old way" which should be deprecated including kyma-operator, kyma-installer & overrides specifics.

AC:
- the "old way" of installation is deprecated
- installation CR is deprecated
- kyma-operator, kyma-installer is deprecated
- overrides are deprecated
- `kyma deploy` is documented
- `kyma undeploy` is documented
- components list is documented
- overrides are documented

**Area**

*Installation

**Reasons**

Documentation should follow installation changes

**Assignees**

@kyma-project/technical-writers

**Attachments**

https://github.com/kyma-project/kyma/issues/10297

### PR DESCRIPTION
**Pull request by mmitoraj**
Updated the installation docs:

Changes proposed in this pull request:

- Added installation prerequisites 
- Added the provisioning step
- Added a reference to the complete list of Kyma components
- Added info about the out-of-the-box kyma.example.com domain

**Related issue(s)**
 Resolves #10355

### CODE REVIEW

**Comment by mmitoraj** [comment]
@strekm @pbochynski Please have a closer look at the **Install Kyma** doc and give your feedback to all the described installation operations, please. Do we still miss anything here?

File: docs/04-operation-guides/operations/01-install-kyma.md [hunk 1]
```diff
@@ -6,7 +6,23 @@ You can simply use the default Kyma installation, or modify it as it fits your p
...
+### Provision and install
+
+You can either use an out-of-the-box k3d cluster or choose any other cluster provider. To quickly provision a k3d cluster run:
+
+  ```bash
+  kyma provision k3s
+  ```
+
+If you use the `deploy` command without any flags, Kyma provides a default domain.
 For example, if you install Kyma on a local cluster, the default URL is `https://console.local.kyma.dev`.
```
Comment (pbochynski) [inline docs/04-operation-guides/operations/01-install-kyma.md:26]: We need to remove console from description. It is not part of Kyma 2.0. The console can be started locally. @valentinvieriu should provide the best option (docker / npx command).

**Review by mjakobczyk — DISMISSED** [review]

File: docs/04-operation-guides/operations/01-install-kyma.md (hunk 1 above)
Comment (mmitoraj) [reply docs/04-operation-guides/operations/01-install-kyma.md:26]: 👍   I'm removing that part and creating a follow-up [issue](https://github.com/kyma-project/kyma/issues/11688) to add the details once we have them.

Comment (mmitoraj) [reply docs/04-operation-guides/operations/01-install-kyma.md]: 👍  I'm removing the line about the Console and creating a [follow-up issue](https://github.com/kyma-project/kyma/issues/11688). TWs will update the doc once we have the required information. @valentinvieriu, I sent you a message on Slack regarding this - we're in touch.

**Review by strekm — DISMISSED** [review]
/meow

File: docs/04-operation-guides/operations/01-install-kyma.md [hunk 2]
```diff
@@ -6,8 +6,26 @@ You can simply use the default Kyma installation, or modify it as it fits your p
...
+
+### Prerequisites
+
+- [Kyma CLI](https://github.com/kyma-project/cli)
+- [Docker](https://docs.docker.com/get-docker/)
+- [k3d](https://k3d.io/#installation)
+
+### Provision and install
+
+You can either use an out-of-the-box k3d cluster or choose any other cluster provider. To quickly provision a k3d cluster run:
```
Comment (NHingerl) [inline docs/04-operation-guides/operations/01-install-kyma.md:19]:
```suggestion
You can either use an out-of-the-box k3d cluster or choose any other cluster provider. To quickly provision a k3d cluster, run:
```

Comment (mmitoraj) [reply docs/04-operation-guides/operations/01-install-kyma.md:19]: 👍

**Review by NHingerl — APPROVED** [review]

**Pull request merged**